app/client/keybase/hashicorp/vault.go

```diff
@@ -0,0 +1,339 @@
+	privKey, err := vk.GetPrivKey(masterAddrHex, passphrase)
+	if err != nil {
+		return nil, err
+	}
+	seed := privKey.Seed()
+	return vk.DeriveChildFromSeed(seed, childIndex, childPassphrase, childHint, shouldStore)
+}
+
+// ExportPrivString exports a private key as a hex string
+func (vk *vaultKeybase) ExportPrivString(address, passphrase string) (string, error) {
+	privKey, err := vk.Get(address)
+	if err != nil {
+		return "", err
+	}
+
+	privKeyHex, err := privKey.ExportString(passphrase)
+	if err != nil {
+		return "", err
+	}
+
+	return privKeyHex, nil
+}
+
+// ExportPrivJSON exports a private key as a JSON string
+func (vk *vaultKeybase) ExportPrivJSON(address, passphrase string) (string, error) {
+	privKey, err := vk.Get(address)
+	if err != nil {
+		return "", err
+	}
+
+	privKeyJSON, err := privKey.ExportJSON(passphrase)
+	if err != nil {
+		return "", err
+	}
+
+	return privKeyJSON, nil
+}
+
+// UpdatePassphrase updates the passphrase of a key
+func (vk *vaultKeybase) UpdatePassphrase(address, oldPassphrase, newPassphrase, hint string) error {
+	privKey, err := vk.GetPrivKey(address, oldPassphrase)
+	if err != nil {
+		return err
+	}
+	privStr := privKey.String()
+
+	keyPair, err := crypto.CreateNewKeyFromString(privStr, newPassphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	return writeVaultKeyPair(vk, keyPair.GetAddressString(), keyPair, hint)
+}
+
+// Sign a message using a private key
+func (vk *vaultKeybase) Sign(address, passphrase string, msg []byte) ([]byte, error) {
+	privKey, err := vk.GetPrivKey(address, passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	sig, err := privKey.Sign(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return sig, nil
+}
+
+// Verify a message signature using a public key
+func (vk *vaultKeybase) Verify(address string, msg, sig []byte) (bool, error) {
+	pubKey, err := vk.GetPubKey(address)
+	if err != nil {
+		return false, err
+	}
+
+	return pubKey.Verify(msg, sig), nil
+}
+
+// Delete a keypair from vault
+func (vk *vaultKeybase) Delete(address, passphrase string) error {
+
+	_, err := vk.GetPrivKey(address, passphrase)
+	if err != nil {
+		return err
+	}
+
+	versionsMeta, err := vk.client.KVv2(vk.mount).GetVersionsAsList(context.TODO(), address)
+	if err != nil {
+		return err
+	}
+
+	versions := make([]int, 0, len(versionsMeta))
+	for _, version := range versionsMeta {
+		versions = append(versions, version.Version)
+	}
+
+	return vk.client.KVv2(vk.mount).Destroy(context.TODO(), address, versions)
+}
+
+// writeVaultKeyPair writes a keypair to vault
+func writeVaultKeyPair(vk *vaultKeybase, address string, kp crypto.KeyPair, hint string) error {
+	dataBz, err := json.Marshal(kp)
+	if err != nil {
+		return err
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), address, map[string]interface{}{
+		"key_pair": string(dataBz),
+		"hint":     hint,
+	})
+
+	return err
+}
```

Comment:
Building on top of the work @h5law did, I honestly think the fact that we have a library for this for easy use with vault is absolute 🔥

Reply:
<img src="https://media.giphy.com/media/3o6Zt2eGSpnHOXeKFq/giphy-downsized.gif" />